[MRG] DOC: Add SLEP and Governance in dev docs

bharatr21 opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->
Reference SLEP and Governance in dev docs as per #13268 

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

Fixes #13268 

#### What does this implement/fix? Explain your changes.
Reference SLEP and Governance  in dev docs

<!-- #### Any other comments? -->


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

bharatr21 committed:
DOC: Add SLEP and Governance in dev docs

jnothman commented on doc/glossary.rst:
```diff
@@ -675,6 +675,18 @@ General Concepts
         See https://scikit-learn-contrib.github.io.
 
+    scikit-learn enhancement proposals
+    SLEPs
```
Let's also add the keyword SLEP

jnothman commented on doc/developers/contributing.rst:
```diff
@@ -40,6 +40,10 @@ ticket to the
+Governance
+----------
+The decision making process and governance structure of scikit-learn is laid
+out in the `governance document <https://github.com/scikit-learn/scikit-learn/blob/master/doc/governance.rst>`_.
```
There should be no need to use a full URL reference. ```:ref:`governance` ``` should suffice

jnothman commented on doc/developers/contributing.rst:
```diff
@@ -57,11 +61,19 @@ investigating bugs, and :ref:`reviewing other developers' pull requests
 <code_review>` are very valuable contributions that decrease the burden on the
 project maintainers.
 
+
```
No need for this blank line

jnothman commented on doc/developers/contributing.rst:
```diff
@@ -57,11 +61,19 @@ investigating bugs, and :ref:`reviewing other developers' pull requests
 
+In case a contribution/issue involves changes to the API principles
+or changes to dependencies or supported versions, a
+`SLEP <https://github.com/scikit-learn/scikit-learn/blob/master/doc/governance.rst#enhancement-proposals-sleps>`_,where a SLEP must be submitted as a pull-request to
```
Please add a link anchor within the governance document and use `:ref:` here

jnothman commented on doc/developers/contributing.rst:
```diff
@@ -57,11 +61,19 @@ investigating bugs, and :ref:`reviewing other developers' pull requests
+`SLEP <https://github.com/scikit-learn/scikit-learn/blob/master/doc/governance.rst#enhancement-proposals-sleps>`_,where a SLEP must be submitted as a pull-request to
+`enhancement_proposals <https://github.com/scikit-learn/enhancement_proposals/>`_
+using the `SLEP template <https://github.com/scikit-learn/enhancement_proposals/blob/master/slep_template.rst>`_
+and follows the decision-making process outlined in the `governance document <https://github.com/scikit-learn/scikit-learn/blob/master/doc/governance.rst>`_.
```
Use `:ref:`

Bharat123rox replied:
Currently could not find/create anchors for the `scikit-learn/enhancement-proposals` folder, so I've retained the full URLs for those cases alone, otherwise requested changes are incorporated.

jnothman reviewed (commented):
Thanks!

bharatr21 committed:
Some ref changes acc to @jnothman

jnothman commented:
https://scikit-learn-enhancement-proposals.readthedocs.io might be
appropriate

bharatr21 committed:
Update enhancement proposal links

bharatr21 committed:
Update enhancement proposal links

DOC Adds recent core devs to _contributors.rst (#13640)

DOC t-SNE perplexity docstring update (#13069)

FIX feature importances in random forests sum up to 1 (#13636)

DOC Removed a typo from the examples of normalized_mutual_info_score (#13646)

MAINT: n_jobs=-1 replaced with n_jobs=4 in tests (#13644)

This change is to work around the hang

 https://github.com/scikit-learn/scikit-learn/issues/12263

afflicting Windows on machines with > 62 hyperthreads.

Add parameter for stacking missing indicator into iterative imputer (#13601)

FIX ignore single node trees in gbm's feature importances (#13620)

DOC typo in sklearn.utils.extmath.weighted_mode (#13655)

1.5+2 = 3.5, not 3.

ENH Extension of v_measure_score metric to include beta parameter (#13607)

BUG Fix missing 'const' in a few memoryview declaration in trees. (#13626)

BLD: check OpenMP support and add a switch to build without it (#13543)

FIX initialise Birch centroid_ in all cases (#13651)

DOC Improve performance of the plot_rbm_logistic_classification.py example (#13383) (#13648)

MNT Import linear assignment from scipy (#13465)

DOC Fixes formatting issue in webkit (#13657)

Fixing parameter description (for assume_centered) (#13456)

* Fixing parameter description (for assume_centered)

Fixing parameter description (for assume_centered)

* Update shrunk_covariance_.py

* Update empirical_covariance_.py

MAINT Unvendor joblib (#13531)

Improve comment in setup.py (#13661)

MAINT Replace absolute imports with relative imports (#13653)

[MRG] Fix various solver issues in ridge_regression and Ridge classes (#13363)

[MRG + 1] Fix pprint ellipsis (#13436)

DOC Remove space in "cross-entropy" (#13671)

FIX broken references in documentation (#13664)

DOC Remove synonyms in documentation of linear models (#13663)

MNT Correctly handle deprecated attribute warnings and docstrings (#13635)

Deprecate "warn_on_dtype" from check_array (#13382)

Fix MultiOutputClassifier checking for predict_proba method of base estimator (#12222)

fix small latex issue (#13680)

Fix sample_weight in label_ranking_average_precision_score (#13447)

DOC Emeritus core devs final call (#13673)

ENH Add verbose option to Pipeline, FeatureUnion, and ColumnTransformer (#11364)

ogrisel commented on doc/glossary.rst:
```diff
@@ -675,6 +675,19 @@ General Concepts
+    SLEPs
+        Changes to the API principles and changes to dependencies or supported
+        versions happen via a `SLEP <governance/#enhancement-proposals-sleps>`_ and follows the
+        decision-making process outlined in the `governance document <governance>`_.
```
Both this link and the previous one are broken in the rendered HTML:

https://54937-843222-gh.circle-artifacts.com/0/doc/glossary.html#term-slep

For the former you probably need to use:

    ... versions happen via a `SLEP <slep>`_ and follows the ...


I do not understand why the second link (targeting the "governance" reference) does not work. The `doc/governance.rst` file correctly includes the following line before its heading:

    .. _governance:

This reference is reachable as expected at:

https://54937-843222-gh.circle-artifacts.com/0/doc/governance.html#governance

and I do not see a typo in the following syntax:

    decision-making process outlined in the `governance document <governance>`_.

Maybe trying to be more explicit with the following would work?

    decision-making process outlined in the :ref:`governance document <governance>`_.

ogrisel replied:
The syntax:

    `governance document <governance>`_

I confirmed it should be:

    :ref:`governance document <governance>`

is actually wrong. I fond other occurrences of this mis-pattern in our documentation. I will issue another PR to fix those.

ogrisel commented on doc/developers/contributing.rst:
```diff
@@ -40,6 +40,10 @@ ticket to the
+Governance
+----------
+The decision making process and governance structure of scikit-learn is laid
+out in the governance document :ref:`governance`.
```
Please insert a colon before the `:ref:` marker:

... out in the governance document: :ref:`governance`.

The `:ref:` marker is replaced by the title of the referenced section in the rendered HTML:

https://54937-843222-gh.circle-artifacts.com/0/doc/developers/contributing.html#governance

ogrisel commented on doc/developers/contributing.rst:
```diff
@@ -62,6 +66,13 @@ up" on issues that others reported and that are relevant to you.  It also helps
+:ref:`slep`, where a SLEP must be submitted as a pull-request to
+`enhancement proposals <https://scikit-learn-enhancement-proposals.readthedocs.io>`_
+using the `SLEP template <https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep_template.html>`_
+and follows the decision-making process outlined in the governance document :ref:`governance`.
```
Here again please please insert a colon before `:ref`.

ogrisel reviewed (changes requested):
I found some issues with the generated links:

bharatr21 committed:
Added changes as proposed by @ogrisel

Bharat123rox commented:
Closing this PR on account of a messed up rebase
